vm-setup/roles/v1aX_integration_test/tasks/remediation.yml

```diff
@@ -220,6 +220,43 @@
     delay: 20
     register: ready_hosts
     until: ready_hosts.stdout == "1"
+    
+    #### m3dt template reference check
+  - name: Get the m3dt,edit and apply a new one 
+    shell: kubectl get m3dt -n "{{ NAMESPACE }}" "{{ CLUSTER_NAME }}"-workers-template  -ojson | 
+              jq '.metadata.name="test-new-m3dt"' |  
+              jq '.|del(.metadata|.managedFields,.uid,.resourceVersion,.ownerReferences,.finalizers)' | 
+              jq '.spec.templateReference="{{ CLUSTER_NAME }}-workers-template"' |
+              kubectl apply -f- 
+
+  - name: Get the m3mt,edit and apply a new one 
+    shell: kubectl get m3mt -n "{{ NAMESPACE }}" "{{ CLUSTER_NAME }}"-workers -ojson |
+              jq '.spec.template.spec.dataTemplate.name="test-new-m3dt"' | 
+              jq '.metadata.name="test-new-m3mt"' |  
+              jq '.|del(.metadata|.managedFields,.uid,.resourceVersion,.ownerReferences,.finalizers)' |
+              kubectl apply -f-
+
+  - name: Edit MachineDeployment to point to the new m3mt
+    shell: kubectl get md -n "{{ NAMESPACE }}" "{{ CLUSTER_NAME }}" -ojson | 
+              jq '.spec.template.spec.infrastructureRef.name="test-new-m3mt"' | 
+              jq '.spec.strategy.rollingUpdate.maxUnavailable=1' |
+              kubectl apply -f-
+
+  - name: Wait until one BMH is in ready state
+    shell: kubectl get bmh -n "{{ NAMESPACE }}" | grep -w ready | wc -l
+    retries: 200
+    delay: 20
+    register: ready_hosts
+    until: ready_hosts.stdout == "1"
+
+  - name: Get the Metal3Data name
+    shell: kubectl get m3d -n "{{ NAMESPACE }}" -o json | jq -r '.items[] | select(.metadata.name | contains("workers"))'| jq .metadata.
+    register: m3d_name
+  
+  - name: Check if Metal3Data refers to the old template
+    shell: "kubectl get m3d -n {{ NAMESPACE }}  {{ m3d_name }} -ojson | jq '.spec.templateReference=={{ CLUSTER_NAME }}-workers-template'"
```

Comment:
what is jq returning ? will the whole command fail if it is not true ? or should there be an ansible exit condition afterwards ?

Reply:
bool. I am not sure, I wanted to check it in CI. But is that check enough or do I need to check something else